CMakeLists.txt

```diff
@@ -220,6 +220,28 @@ install(
   COMPONENT dev)
 
 find_package(CycloneDDS REQUIRED)
+# Support for shared memory in Cyclone DDS C++ depends on support for shared
+# memory being compiled into Cyclone DDS and iceoryx_binding_c being available
+# If Cyclone DDS is compiled with support for shared memory, Cyclone DDS C++
+# must be compiled with shared memory too and vice versa.
+get_target_property(cyclonedds_has_shm CycloneDDS::ddsc SHM_SUPPORT_IS_AVAILABLE)
+mark_as_advanced(cyclonedds_has_shm)
+
+option(ENABLE_SHM "Enable shared memory support" ${cyclonedds_has_shm})
+if(ENABLE_SHM)
+  if(NOT cyclonedds_has_shm)
+    message(FATAL_ERROR "Cyclone DDS is not compiled with support for shared memory")
+  else()
+    find_package(iceoryx_binding_c REQUIRED)
+  endif()
+  message(STATUS "Found iceoryx_binding_c: (found version \"${iceoryx_binding_c_VERSION}\")")
+  set(ENABLE_SHM ON)
+  set(DDSCXX_HAS_SHM "1")
+else(cyclonedds_has_shm)
```

Comment:
@k0ekk0ek Probably this is the reason for the build failure! If Cyclone DDS is not build wioth SHM support then we always throw this fatal error here, which is incorrect, no?

Reply:
Yeah, caught it. That's indeed correct :+1:. Look out for another PR. I've also added a pipeline build, which I'm testing now.

Reply:
PR: https://github.com/ApexAI/cyclonedds-cxx/pull/2